File: Roadmap/00 - SINTAXIS, VARIABLES, TIPOS DE DATOS Y HOLA MUNDO/java/BertoMP.java
public class BertoMP {
    public static void main(String[] args) {
        // WEB OFICIAL
        // Para encontrar información sobre Java dirígite a su web oficial: https://www.java.com/

        // COMENTARIOS
        // En Java tenemos 3 tipos de comentarios:

        // - Comentarios de una única línea (// Comentario):
        // Sirven por ejemplo para explicar una variable.

        // - Comentarios de varias líneas (/* Comentario */):
        /* Este tipo de comentarios sirven
         * para explicar por ejemplo un pequeño
         * proceso que ocurre dentro de una
         * de nuestras funciones. */

        // - Comentarios de documentación (/** Comentario */):
        /**
         * Este tipo de comentarios se utilizan para generar documentación
         * sobre el código que en el caso de Java se denomina JavaDoc.
         * Por ejemplo, los podremos ver explicando de forma detallada cómo
         * funciona una de las funciones que hayan sido implementadas.
         * Suelen ir acompañadas de etiquetas que ayudan a entender mejor
         * el código o aportan información extra:
         *  - @param: Determina el tipo de parámetro que recibe la función.
         *  - @return: Determina el tipo de resultado/retorno que produce
         *             la función al final de su ejecución.
         *  - @version: Versión del método o la clase.
         *  - Etc.
         */

        // VARIABLES Y CONSTANTES
        /* Para declarar una variable en Java debemos indicar su tipo y el
         * nombre de la variable. Puede estar inicializada o no, en el caso
         * de que no lo esté tendrá un valor por defecto u otro.
         * Por convención, el nombre de las variables debe seguir la nomenclatura
         * camelCase. */
        int miVariableInicializada = 33;
        int miVariableNoInicializada;

        /* Para declarar una constante en Java debemos hacer algo similar
         * pero utilizando la palabra reservada 'final'. Además, todas
         * las constantes deben ser inicializadas en su declaración y,
         * por convención, es recomendable que su nombre sea en mayúsculas. */
        final int MI_CONSTANTE = 40;

        // DATOS PRIMITIVOS
        // En Java tenemos los siguientes tipos de datos primitivos:

        // - DATOS NUMÉRICOS ENTEROS
        byte miByte = 12; // Números desde -128 hasta 127.
        short miShort = 12642; // Números desde -32.768 hasta el 32.767.
        int miInt = 1726171821; // Números desde -2.147.483.648 hasta 2.147.483.647.
        long miLong = 10000000000L; // Números desde -9.223.372.036.854.775.808 hasta 9.223.372.038.854.775.807.

        // - DATOS NUMÉRICOS DECIMALES
        float miFloat = 3.141592f; // Números hasta 6-7 decimales.
        double miDouble = 3.141592653; // Números hasta 15 decimales.

        // - DATOS LÓGICOS
        boolean miBoolean = true; // Almacena un valor booleano (true o false).

        // - CARACTERES
        char miChar = 'A'; // Almacena un carácter o un valor ASCII.

        /* Para declarar cadenas de texto en Java no tenemos un tipo primitivo como tal,
         * sino que se utiliza la clase String para ello. */
        String miString = "Esto es un String de prueba";

        // IMPRESIÓN POR CONSOLA
        System.out.println("¡Hola, Java!");
    }
}

File: Roadmap/00 - SINTAXIS, VARIABLES, TIPOS DE DATOS Y HOLA MUNDO/php/BertoMP.php
<?php
// WEB OFICIAL
// Sitio web oficial de PHP: https://www.php.net/

// COMENTARIOS
// En PHP tenemos 3 tipos de comentarios:

// - Comentarios de una sola línea (// Comentario): Se utilizan para explicar partes concretas del código.
// Esto es un comentario de una línea.

// - Comentarios de varias líneas (/* Comentario */): Estos comentarios permiten explicar bloques extensos de código.
/*
 * Esto es un ejemplo de
 * un comentario
 * de varias líneas. */

// - Comentarios de documentación (/** Comentario */): Sirven para generar documentación.
/**
 * Este tipo de comentarios son los que usaremos cuando queramos
 * explicar el funcionamiento de, por ejemplo, una función
 * o una clase de PHP.
 * Generan documentación de forma automática, también llamada
 * PHPDoc y suelen ir acompañados de etiquetas que mejoran
 * su comprensión:
 * - @param: Nos sirve para determinar el tipo de variable que
 *           recibe la función.
 * - @return: En este caso lo que nos informará del tipo de
 *            retorno o resultado que produce la función. */

// VARIABLES Y CONSTANTES
/* - Variables: Para declarar una variable en PHP utilizamos
                el nombre que le queramos dar a la variable
                precedido por el símbolo del dólar ($).
                Podemos dejar una variable sin inicializar o
                darle el valor null. */
$miVariable = 12;
$miVariableSinInicializar = null;

/* - Constantes: En PHP tenemos dos formas de declarar una constante,
                 la primera de ellas es la palabra const seguido del
                 nombre de nuestra constante y su valor.
                 También las podemos declarar con el método define(),
                 que recibe 2 parámetros: un string con el nombre de la
                 constante y un valor.
                 Por convención, las constantes deben declararse con mayúsculas. */
const MI_CONSTANTE = 33;
define('MI_OTRA_CONSTANTE', 40);

// DATOS PRIMITIVOS
// En PHP tenemos los siguientes tipos de datos primitivos:
$miInteger = 12; // Integer: Almacena números enteros.
$miFloat = 12.64; // Float: Almacena números con decimales.
$miBoolean = true; // Boolean: Sirve para valores lógicos (true o false).
$miNull = null; // Null: Almacena un dato nulo -sin valor-.
$miString = 'Esto es una cadena'; // String: Sirve para cadenas de texto o caracteres únicos.
$miArray = array('a', 'b', 'c'); // Array: Sirve para almacenar varios datos en un mismo lugar.

// IMPRESIÓN
echo '¡Hola, PHP!';